File: 2022/10/21/21.ts
(() => {
    const numbers: number[] = [1, 2, 3, 4, 5];

    // Without using division
    const getProductNumbers = (numbers: number[]): number[] => {
        let productNumbers: number[] = [];
        for (let number of numbers) {
            let product = 1;
            for (let number2 of numbers) {
                if (number !== number2) product *= number2;
            }
            productNumbers.push(product);
        }
        return productNumbers;
    }

    const getProductNumbersAlternative = (numbers: number[]): number[] => {
        let product = 1;
        for (let number of numbers) product *= number;
        return numbers.map(number => product / number);
    }

    console.log(getProductNumbers(numbers));
    console.log(getProductNumbersAlternative(numbers));
})();

File: 2022/10/10-21-2022/21.py
numbers_given: list = [1, 2, 3, 4, 5]


# def get_product_numbers(numbers: list) -> list:
#     total_product: int = 1
#     for number in numbers:
#         total_product *= number
#     return [total_product // number for number in numbers]


def get_product_numbers(numbers: list) -> list:
    product_numbers = []
    for number in numbers:
        total = 1
        for other_number in numbers:
            if number != other_number:
                total *= other_number
        product_numbers.append(total)
    return product_numbers


print(get_product_numbers(numbers_given))
